### ISSUE TITLE
`cppcoreguidelines-rvalue-reference-param-not-moved` not reported for templated method

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#61420]
```cpp
#include <string>

class C
{
public:
    template<typename T>
    void str(T&& s) { // no warning
        mStr = s;
    }

    void str2(std::string&& s) { // warning
        mStr = s;
    }
private:
    std::string mStr;
};

void f()
{
    C c;
    c.str("");
    c.str2("");
}
```

```console
<source>:11:29: warning: rvalue reference parameter 's' is never moved from inside the function body [cppcoreguidelines-rvalue-reference-param-not-moved]
    void str2(std::string&& s) {
                            ^
```

https://godbolt.org/z/zddq9hz34

**Comment by PiotrZSL** [comment]
No issue here:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f18-for-will-move-from-parameters-pass-by-x-and-stdmove-the-parameter

Enforcement: Flag all X&& parameters (where X is not a template type parameter name) where the function body uses them without std::move.
If someone would use this:
```
    c.str(std::string());
    std::string s;
    c.str(s);
```

Then you got problem, because `std::move` could be invalid. You probably looking for some other check like one that suggest using `std::forward`:
F.19: For “forward” parameters, pass by TP&& and only std::forward the parameter
But currently check for F.19 is not implemented

**Comment by firewave** [comment]
There's already https://clang.llvm.org/extra/clang-tidy/checks/bugprone/move-forwarding-reference.html which is somehow related.

**Comment by ccotter** [comment]
https://reviews.llvm.org/D146921 implements cppcoreguideline F.19, which flags when a forwarding reference is not forwarded with `std::forward` in the function body.

**Comment by PiotrZSL** [comment]
Already fixed by new check.

**Comment by firewave** [comment]
I assume the new check is `cppcoreguidelines-missing-std-forward`.
https://godbolt.org/z/5KMhn9Err

 A reference to the commit fixing this would be great if closing the ticket.

**Comment by firewave** [comment]
Your comment made it seem like it was different from the referenced review. My bad.
